view/user.py

```diff
@@ -99,3 +104,46 @@ def sign_up():
             res['message'] = 'sign up successful'
         return jsonify(**res)
     return render_template('index.html')
+
+@app.route('/profile', methods=['GET', 'POST'])
+def profile():
+    res={}
+    res['code'] = '0x1'
+    res['message'] = 'modify profile successful'
+    if request.method == 'POST':
+        eagle_logger.debug(type(request.data))
+        req_data = json.loads(request.data)
+        db_session = db.Session()
+        testid = req_data['id']
+        result = db_session.query(User).filter(User.id == testid).first()
+        req_id = result.id
+        new_username = req_data['username']
+        if new_username is not None:
+            db_session = db.Session()
+            result_User = db_session.query(User).filter(User.username == new_username).first()
+            if result_User is not None and req_id != result_User.id:
+                res['code'] = '0x4'
+                res['message'] = 'Username has been occupied by others'
+                return jsonify(**res)
+            else:
+                update_username_by_id(req_id, new_username)
+        new_email = req_data['email']
+        if new_email is not None:
+            db_session = db.Session()
+            result_User = db_session.query(User).filter(User.email == new_email).first()                   
+            if result_User is not None and req_id != result_User.id:
+                res['code'] = '0x5'
+                res['message'] = 'Email has been occupied by others'
+                return jsonify(**res)
+            else:
+                update_email_by_id(req_id, new_email)
```

Comment:
If the username is not occupied by others, but the email is occupied by others.
The username is modified in database.
Maybe it's not suiltable.